library/psa_crypto.c

```diff
@@ -1142,7 +1142,7 @@ static psa_status_t psa_get_key_from_slot( psa_key_handle_t handle,
 
     *p_slot = NULL;
 
-    status = psa_get_key_slot( handle, &slot );
+    status = psa_get_key_slot( key, &slot );
     if( status != PSA_SUCCESS )
```

Comment:
I'm not sure what you mean here. If the key identifier is not correct, there is nothing to remove. In case of failure for internal reasons (insufficient memory for example) maybe but I would say such error handling shouldn't be here.

Reply:
Ok: `psa_get_key_slot` cleans up after itself — it won't return with a partially allocated slot. So there's no need for any cleanup here in `psa_get_key_from_slot`.